languages/julia/exercises/concept/multiple-dispatch/.meta/config.json

```
@@ -4,5 +4,7 @@
             "github_username": "SaschaMann",
             "exercism_username": "SaschaMann"
         }
-    ]
+    ],
+    "supported_language_versions": "≥1.0",
```

Can we make this semver?

```suggestion
    "language_version": "1.0",
```

---

I don't think all languages follow semver for their versions.

---

Well, I assume that their version numbers will increment and they could then just use the major version.

---

Can we either use:

```json
{
   "language_version": { "semver": "~> 1.0" }
   "language_version": { "ver": "1.0" }
}

{
   "language_version": { "semver": true, "version": "~> 1.0" }
   "language_version": { "semver": false, "version": "1.0" }
}

{
   "language_sem_version": { "~> 1.0" }
   "language_version":  "1.0" }
}
```

For tooling _knowing_ which ones are semver is very helpful.

---

(each group shows both options. They're not meant to be used at the same time)

---

Some, like C or Fortran don't even have x.y.z versions. Would a free-form version string cause any issues?

---

Not if I can tell which ones are semver.

---

I'm fine with either of those suggestions. @ErikSchierboom which would you prefer?

---

> For tooling knowing which ones are semver is very helpful.

What tooling are you referring to?

---

As erik hints at, I think the question I have is what is the actual purpose of this field? That should determine it's format, but I'm not clear on what is expected to consume this yet.

---

It should show a hint on the website which version is required for the student to solve the exercise.

---

Will this language version field be optional? Common Lisp only has one version - an ANSI specification (ANSI INCITS 226-1994 (R2004)). So not relevant at all for that language.

---

I want people to be able to get a warning locally if they are on an unsupported version. It's important for Ruby for example. Not so much for JavaScript at this moment because use the transpiler --

---

I'm okay with using a single version field for that purpose -- but I can imagine some more generic tooling at a higher level 🗡

---

@verdammelt Yep, this definitely will be an optional. Shall we go for `"language_version": "1.0",
` then? Personally, I feel that the [NPM semantic version format](https://docs.npmjs.com/about-semantic-versioning) is very nice and flexible enough to handle various scenarios. It is also quite widely known and will have tooling for it. @SaschaMann what do you think about using the NPM semantic version format?

---

> It should show a hint on the website which version is required for the student to solve the exercise. In v2 it's just a string in the README.
> I want people to be able to get a warning locally if they are on an unsupported version. It's important for Ruby for example. 

These are two very different use cases. The one is something that appears verbatim to students, the other is something machine-readable used to check something.

Does this difference warrant separate fields?

---

Could be. @SleeplessByte?

---

It seems to me like the simplest and most reliable thing would be to make this a free-form string field, and allow language track maintainers to create tooling which can check the language against this string in the appropriate way.

---

@ErikSchierboom I find it rather obscure[1], and it doesn't work for every language. I just want to let the student know which version of the language they need to use. I agree with @bitfield that the easiest solution would be a free-form string.

[1] Unless you're familiar with semver versions in general, and the npm format in particular, you won't understand what it means. `1.0` reads like it requires version `1.0` but `1.1` might not work. `≥1.0` or `1.0 or later` or w/e is understandable without requiring that knowledge.